### PR TITLE
Finished test

### PR DESCRIPTION
**Pull request by Parthib**
These are the public test cases from the Android tutorial from Stan

### CODE REVIEW

File: .idea/gradle.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
       <GradleProjectSettings>
         <option name="distributionType" value="LOCAL" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleHome" value="$APPLICATION_HOME_DIR$/gradle/gradle-2.14.1" />
+        <option name="gradleHome" value="C:\Program Files\Android\Android Studio\gradle\gradle-2.14.1" />
```
Comment (stswidwinski) [inline .idea/gradle.xml]: Directory shouldn't be hardcoded! :)

File: .idea/misc.xml [hunk 2]
```diff
@@ -37,7 +37,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (stswidwinski) [inline .idea/misc.xml]: No JDK downgrade

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        mMockContext = Mockito.mock(Context.class, Mockito.RETURNS_DEEP_STUBS);
+
+        DisplayMetrics dm = Mockito.mock(DisplayMetrics.class, Mockito.RETURNS_DEEP_STUBS);
+
+        dm.widthPixels = 50;
+        dm.heightPixels = 100;
+
+        Mockito.when(mMockContext.getResources().getDisplayMetrics()).thenReturn(dm);
+
```
Comment (stswidwinski) [inline app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java:37]: Spacing.

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @Mock
+    Context mMockContext;
+
+    @Test
+    public void getCacheSize_isCorrect() throws Exception {
+
+        mMockContext = Mockito.mock(Context.class, Mockito.RETURNS_DEEP_STUBS);
+
+        DisplayMetrics dm = Mockito.mock(DisplayMetrics.class, Mockito.RETURNS_DEEP_STUBS);
```
Comment (stswidwinski) [inline app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java]: Why do you need deep stubbing or stubbing at all on DisplayMetrics?

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java (hunk 3 above)
Comment (lasyasreepada) [inline app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java:37]: Add spaces between the operands in your 'equalTo' clause, e.g (dm.widthPixels * dm.heightPixels * 4 * 3)

Comment (Parthib) [reply app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java:37]: Fixed

Comment (Parthib) [reply app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java:37]: Fixed

**Pull request closed without merging**